### PR TITLE
Backport changes from 1.21.1

### PR DESCRIPTION
**Pull request by IThundxr**

### CODE REVIEW

File: buildSrc/src/main/kotlin/dev/engine_room/gradle/transitive/TransitiveSourceSetsExtension.kt [hunk 1]
```diff
@@ -25,7 +26,7 @@ open class TransitiveSourceSetsExtension(private val project: Project) {
 
     fun createCompileConfigurations() {
         val configs = transitives.mapValues { (sourceSet, _) ->
-            project.configurations.create("for${sourceSet.name.capitalize()}") {
+            project.configurations.create("for${sourceSet.name.replaceFirstChar { it.uppercase() }}") {
```
Comment (PepperCode1) [inline buildSrc/src/main/kotlin/dev/engine_room/gradle/transitive/TransitiveSourceSetsExtension.kt:28]: Why was this change made?

File: common/src/main/java/dev/engine_room/flywheel/impl/compat/SodiumCompat.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package dev.engine_room.flywheel.impl.compat;
+
+import dev.engine_room.flywheel.lib.visualization.VisualizationHelper;
+import net.caffeinemc.mods.sodium.api.blockentity.BlockEntityRenderHandler;
+import net.caffeinemc.mods.sodium.api.blockentity.BlockEntityRenderPredicate;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+
+public class SodiumCompat {
+	public static <T extends BlockEntity> Object forBlockEntityType(BlockEntityType<T> type) {
```
Comment (PepperCode1) [inline common/src/main/java/dev/engine_room/flywheel/impl/compat/SodiumCompat.java]: Rename to `addPredicate` for consistency with the other method.

File: common/src/main/java/dev/engine_room/flywheel/impl/mixin/BlockEntityTypeMixin.java [hunk 3]
```diff
@@ -5,15 +5,22 @@
...
+import dev.engine_room.flywheel.impl.compat.CompatMods;
+import dev.engine_room.flywheel.impl.compat.SodiumCompat;
 import dev.engine_room.flywheel.impl.extension.BlockEntityTypeExtension;
 import net.minecraft.world.level.block.entity.BlockEntity;
 import net.minecraft.world.level.block.entity.BlockEntityType;
 
 @Mixin(BlockEntityType.class)
 abstract class BlockEntityTypeMixin<T extends BlockEntity> implements BlockEntityTypeExtension<T> {
 	@Unique
+	@Nullable
```
Comment (PepperCode1) [inline common/src/main/java/dev/engine_room/flywheel/impl/mixin/BlockEntityTypeMixin.java:16]: Also add `@Nullable` to the equivalent field in `EntityTypeMixin`.

File: forge/src/main/java/dev/engine_room/flywheel/impl/FlywheelForge.java [hunk 4]
```diff
@@ -63,6 +65,9 @@ private static void clientInit(IEventBus forgeEventBus, IEventBus modEventBus) {
 
 		CrashReportCallables.registerCrashCallable("Flywheel Backend", BackendManagerImpl::getBackendString);
 		FlwImpl.init();
+
+		if (CompatMods.EMBEDDIUM.isLoaded())
```
Comment (PepperCode1) [inline forge/src/main/java/dev/engine_room/flywheel/impl/FlywheelForge.java]: Use braces.

File: forge/src/main/resources/META-INF/mods.toml [hunk 5]
```diff
@@ -26,10 +26,16 @@ mandatory = true
 versionRange = "${forge_version_range}"
 side = "CLIENT"
 
+[[dependencies.flywheel]]
```
Comment (PepperCode1) [inline forge/src/main/resources/META-INF/mods.toml]: Use `${mod_id}` instead of `flywheel`.

**Review by PepperCode1 — CHANGES_REQUESTED** [review]
After addressing review comments, also organize imports in all changed files as some are out of order.

File: buildSrc/src/main/kotlin/dev/engine_room/gradle/transitive/TransitiveSourceSetsExtension.kt (hunk 1 above)
Comment (IThundxr) [reply buildSrc/src/main/kotlin/dev/engine_room/gradle/transitive/TransitiveSourceSetsExtension.kt:28]: capitalize is deprecated, it shouldn't be used anymore

File: forge/src/main/java/dev/engine_room/flywheel/impl/mixin/sodium/ChunkBuilderMeshingTaskMixin.java
Comment (Jozufozu) [inline forge/src/main/java/dev/engine_room/flywheel/impl/mixin/sodium/ChunkBuilderMeshingTaskMixin.java:1]: Don't we still need these on 1.20.1? Is sodium 0.6 getting backported?

Comment (IThundxr) [reply forge/src/main/java/dev/engine_room/flywheel/impl/mixin/sodium/ChunkBuilderMeshingTaskMixin.java:1]: Right...

Comment (IThundxr) [reply forge/src/main/java/dev/engine_room/flywheel/impl/mixin/sodium/ChunkBuilderMeshingTaskMixin.java:1]: It's not needed on forge as there's proper embeddium compat, but on fabric it's still needed and it needs to be controlled by the version of sodium that is loaded

**Review by PepperCode1 — APPROVED** [review]

**Review by Jozufozu — APPROVED** [review]

**Pull request merged**